ipmanager/hetznerFloatingIpConfigurer.go

```diff
@@ -0,0 +1,193 @@
+package ipmanager
+
+import (
+	"context"
+	"errors"
+	"log"
+	"time"
+
+	"github.com/cybertec-postgresql/vip-manager/vipconfig"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+)
+
+type HetznerFloatingIpConfigurer struct {
+	*IPConfiguration
+	client       *hcloud.Client
+	server       *hcloud.Server
+	floatingIpId string
+	cachedState  int
+	lastAPICheck time.Time
+	verbose      bool
+}
+
+func newHetznerFloatingIpConfigurer(config *vipconfig.Config, ipConfig *IPConfiguration) (*HetznerFloatingIpConfigurer, error) {
+	client := hcloud.NewClient(hcloud.WithToken(config.HetznerCloudToken))
+	if client == nil {
+		return nil, errors.New("Failed to connect to hetzner cloud")
+	}
+
+	server, _, err := client.Server.Get(context.Background(), config.HetznerCloudServerId)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Please put this into its own method so that we can recreate this whenever access to the API fails.